bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java

```diff
@@ -0,0 +1,385 @@
+    private ThingRegistry thingRegistry;
+    private String globeType = "";
+    private String bulbMode = "";
+    private String remotesGroupID = "";
+    private String channelPrefix = "";
+    private String fullCommandTopic = "";
+    private String fullStatesTopic = "";
+    private BigDecimal maxColourTemp = BigDecimal.ZERO;
+    private BigDecimal minColourTemp = BigDecimal.ZERO;
+    private BigDecimal savedLevel = BIG_DECIMAL_100;
+    private ConfigOptions config = new ConfigOptions();
+
+    public EspMilightHubHandler(Thing thing, ThingRegistry thingRegistry) {
+        super(thing);
+        this.thingRegistry = thingRegistry;
+    }
+
+    void changeChannel(String channel, State state) {
+        updateState(new ChannelUID(channelPrefix + channel), state);
+        // Remote code of 0 means that all channels need to follow this change.
+        if (remotesGroupID.equals("0")) {
+            switch (globeType) {
+                // These two are 8 channel remotes
+                case "fut091":
+                case "fut089":
+                    updateState(new ChannelUID(channelPrefix.substring(0, channelPrefix.length() - 2) + "5:" + channel),
+                            state);
+                    updateState(new ChannelUID(channelPrefix.substring(0, channelPrefix.length() - 2) + "6:" + channel),
+                            state);
+                    updateState(new ChannelUID(channelPrefix.substring(0, channelPrefix.length() - 2) + "7:" + channel),
+                            state);
+                    updateState(new ChannelUID(channelPrefix.substring(0, channelPrefix.length() - 2) + "8:" + channel),
+                            state);
+                default:
+                    updateState(new ChannelUID(channelPrefix.substring(0, channelPrefix.length() - 2) + "1:" + channel),
+                            state);
+                    updateState(new ChannelUID(channelPrefix.substring(0, channelPrefix.length() - 2) + "2:" + channel),
+                            state);
+                    updateState(new ChannelUID(channelPrefix.substring(0, channelPrefix.length() - 2) + "3:" + channel),
+                            state);
+                    updateState(new ChannelUID(channelPrefix.substring(0, channelPrefix.length() - 2) + "4:" + channel),
+                            state);
+            }
+        }
+    }
+
+    private void processIncomingState(String messageJSON) {
+        // Need to handle State and Level at the same time to process level=0 as off//
+        BigDecimal tempBulbLevel = BigDecimal.ZERO;
+        String bulbState = Helper.resolveJSON(messageJSON, "\"state\":\"", 3);
+        String bulbLevel = Helper.resolveJSON(messageJSON, "\"level\":", 3);
+        if (!bulbLevel.isEmpty()) {
+            if (bulbLevel.equals("0") || bulbState.equals("OFF")) {
+                changeChannel(CHANNEL_LEVEL, new PercentType(0));
+                changeChannel(CHANNEL_LEVEL, OnOffType.OFF);
```

Comment:
Why are you updating the same channel twice?

Reply:
Fixed, no reason to do it twice.